match/match_it_test.go

```diff
@@ -0,0 +1,205 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/TempleEight/spec-golang/match/comm"
+	"github.com/TempleEight/spec-golang/match/dao"
+	"github.com/TempleEight/spec-golang/match/util"
+)
+
+const dateTimeFormat = "2006-01-02T15:04:05.000000Z"
+
+var environment env
+
+func TestMain(m *testing.M) {
+	config, err := util.GetConfig("/etc/match-service/config.json")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	d, err := dao.Init(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+	c := comm.Init(config)
+
+	environment = env{d, c}
+
+	// Create two users for the test
+	url := config.Services["user"]
+	client := http.Client{
+		Timeout: 5 * time.Second,
+	}
+
+	req, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer([]byte(`{"Name": "Jay"}`)))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	req.Header.Set("Authorization", "Bearer "+JWT0)
+	_, err = client.Do(req)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	req, err = http.NewRequest(http.MethodPost, url, bytes.NewBuffer([]byte(`{"Name": "Lewis"}`)))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	req.Header.Set("Authorization", "Bearer "+JWT1)
+	_, err = client.Do(req)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	os.Exit(m.Run())
+}
+
+func TestIntegrationMatch(t *testing.T) {
```

Comment:
Might be worth thinking about using private tests as part of the mega test, though not sure how well that will work with logging of lines etc

    func TestIntegrationMega() {
        setUpStuff...
        testPartOne();
        testPartTwo();
        testPartThree();
    }